packages/astro/src/core/build/static-build.ts

```diff
@@ -161,7 +155,7 @@ export async function staticBuild(
 			settings.timer.end('Server generate');
 			return;
 		}
-		default:
+		default: // `settings.buildOutput` will always be one of the above, but TS doesn't know that
```

Comment:
Isn't it a union of the two strings? TypeScript should know that, and not need the default. See https://www.typescriptlang.org/play/?#code/C4TwDgpgBAzg9gWwsAFgSwHYHMoF4oDkcGEBUAPocAO5wEDcAUIwGYCuGAxsGsVACZwAysDYsWAClSYsALliJk6bAEp5GNggBGEAE5QA3oygnY1NME4opyrCsPHTTzgEMY0IiQLzdyNrowoAEYmJ2c3Dxo6Hz8AqAAmUNMAX0ZkoA

Reply:
It's not, the default value is `undefined`, for the short instant where we don't know yet the output.